### DISCUSSION TITLE
What is the ideal solution working with regular brew installs and casks installs (sudo permissions)?

### DISCUSSION THREAD

**Discussion by waliddoch** [Homebrew/discussions#965] [Casks]
I'm sure this isn't only my issue (macOS):
Having 2 users - an admin user with sudo permissions, and a regular user (in terms of better security) without sudo permissions. 

the ideal way getting this to work is to do everything through the regular user (that has no sudo permissions). Unfortunately Some cask installations do not allow that.
 
I want to install python@3.9 for example which works great, but the problems occur when installing some cask apps like bartender - that require sudo, which I cannot do using the regular user (it would ask for a password, using the regular user and the admin user passwords do not work). 

ok, so I give my admin user permissions for files that brew requires, but then I can only install stuff only if I log in as the admin user (in the terminal `su Wadmin`) and enter the password.. 

but then get some problems updating apps that were installed using my regular user, and that aside if I decide to use my sudo user for brew - I'm not sure I want python installed through a super user (this has some consequences right?)  

I tried managing both in separate ways, i log in to my sudo user for all cask installs, and for the regular installs I just use the regular user - then when running `brew upgrade` i get things broken down.    

What is the common solution for this situation? 
what is the best way to manage brew and brew user(/s) permissions?

**Comment by SMillerDev** [comment]
> What is the common solution for this situation?

"If you need to run Homebrew in a multi-user environment, consider creating a separate user account especially for use of Homebrew."
~ https://docs.brew.sh/FAQ#why-does-homebrew-say-sudo-is-bad

> what is the best way to manage brew and brew user(/s) permissions?

The best way is to install everything under a single user since that is what brew was designed for. The second best way is to make a brew user to do all your brew tasks as described above.

**Reply by waliddoch** [reply]
Hello @SMillerDev and thanks for your reply. 
I'm not sure you understood the problem. I mean, what's the point of having another user, which too - cannot use sudo. 
When using cask, some (actually, most) apps require sudo privileges, so this won't resolve my problem since brew doesn't allow or recommend using sudo (as said in the link you pasted). 
I was wondering if there is a solution for this, or at least a propper workaround.

**Reply by SMillerDev** [reply]
Since casks just run the installer that a project provides there is nothing brew can do about them requiring admin privileges.

**Reply by waliddoch** [reply]
> Since casks just run the installer that a project provides there is nothing brew can do about them requiring admin privileges.

and no workaround solutions? 
This is pretty lame, because it means that I cannot install apps that require sudo, and I definetly do not want to install apps under a user with sudo prevliges :/

**Reply by SMillerDev** [reply]
You can request with the project that you're trying to install if they can make the installer available without requiring administrator privileges.

**Reply by waliddoch** [reply]
> You can request with the project that you're trying to install if they can make the installer available without requiring administrator privileges.

I think I have found a workaround to this problem but I'm not sure what's the consequence of it. 
I add `--appdir='/Applications'` - for example:  `brew install vlc --appdir='/Applications'`

Any idea what's the difference between this to the regular way without the `--appdir=`?

**Reply by Kentzo** [reply]
You're altering the installation of the application (assuming it's app only). As long as you specify the dir that the user running brew can write to then you're good to go.
